spv/sync.go

```diff
@@ -704,6 +705,63 @@ func (s *Syncer) receiveInv(ctx context.Context) error {
 	}
 }
 
+// receiveInitState receives initial state messages from peers
+// and starts goroutines to handle tspend hashes. Requests all
+// unseen tspend txs and adds them to the tspends cache.
+func (s *Syncer) receiveInitState(ctx context.Context) error {
+	const opf = "spv.receiveInitState(%v)"
+	var wg sync.WaitGroup
+	for {
+		rp, msg, err := s.lp.ReceiveInitState(ctx)
+		if err != nil {
+			wg.Wait()
+			return err
+		}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			unseenTSpends := make([]*chainhash.Hash, 0)
+			for h := range msg.TSpendHashes {
+				if !s.wallet.IsTSpendCached(&msg.TSpendHashes[h]) {
+					unseenTSpends = append(unseenTSpends, &msg.TSpendHashes[h])
+				}
+			}
+
+			if len(unseenTSpends) == 0 {
+				return
+			}
+
+			tspendTxs, err := rp.Transactions(ctx, unseenTSpends)
+			if errors.Is(err, errors.NotExist) {
+				err = nil
+				// Remove notfound txs.
+				prevTxs, prevUnseen := tspendTxs, unseenTSpends
+				tspendTxs, unseenTSpends = tspendTxs[:0], unseenTSpends[:0]
+				for i, tx := range prevTxs {
+					if tx != nil {
+						tspendTxs = append(tspendTxs, tx)
```

Comment:
You need to double check the received transaction is actually a tspend by calling `stake.IsTSpend`, and ideally also check the signature is valid for tspend keys for the current network and that it has not expired.